provisioning/security/tpm-provider/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMHsm.java

```diff
@@ -135,6 +134,8 @@ private TPMT_PUBLIC createPersistentPrimary(Tpm tpm, TPM_HANDLE hPersistent, TPM
             throw new SecurityProviderException("Unexpected failure {" +  rc.name() + "} of TPM2_ReadPublic for {" + primaryRole + "}");
         }
 
+        log.info("Creating {} since the TPM didn't have one already", primaryRole);
```

Comment:
Q - Wouldn't a hardware tpm device always come with an embedded ek? Is it even relevant for us to be creating an ek for an hsm device?

Reply:
In practice, yes, hardware TPMs are expected to have an EK. In theory though, it isn't required. I don't expect most users to hit this code path

Reply:
To add additional context, TPMs support clearing EKs and replacing them with new EKs. That's why this code technically worked without these changes. It's just an unusual case.

Reply:
That's true, I wouldn't expect this code path to be exercised with hsm flow; no harm in leaving it here though